test/unit/DataLoader.test.js

```diff
@@ -0,0 +1,165 @@
+/* eslint-env mocha */
+import React from 'react';
+import sinon from 'sinon';
+import axios from 'axios';
+import { expect } from 'chai';
+import Enzyme, { shallow } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+import DataLoader from './../../src/app/components/DataLoader/DataLoader';
+import Actions from './../../src/app/actions/Actions';
+import appConfig from '@appConfig';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe('DataLoader', () => {
+  describe('Non-matching path', () => {
+    let component;
+    let loadingAction;
+    let bibAction;
+    let axiosStub;
+    const axiosCalls = [];
+    const location = {
+      pathname: '/research/collections/shared-collection-catalog/nonMatchingPath',
+    };
+
+    before(() => {
+      loadingAction = sinon.spy(Actions, 'updateLoadingStatus');
+      bibAction = sinon.spy(Actions, 'updateBib');
+
+      axiosStub = sinon.stub(axios, 'get').callsFake(args =>
+        new Promise((resolve) => {
+          axiosCalls.push(args);
+          resolve({ data: 'bib' });
+        }),
+      );
+    });
+
+
+    after(() => {
+      loadingAction.restore();
+      bibAction.restore();
+      axiosStub.restore();
+    });
+
+    it('should not call axios', () => {
+      expect(axiosCalls.length).to.equal(0);
+    });
+
+    it('should make no Action calls', () => {
+      expect(loadingAction.called).to.equal(false);
+      expect(bibAction.called).to.equal(false);
+    });
```

Comment:
You're not mounting the component so these aren't really being tested.

Reply:
Thanks, fixed!